### ISSUE TITLE
Matrix4 on IOS with impeller is not working

### ISSUE THREAD

**Issue by ManInTheWind** [flutter/flutter#120589]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
the container doesnt render out and without any perspective effect
![image](https://user-images.githubusercontent.com/32859126/218433772-ff4c5570-7e9b-4249-b52e-2ecffc6d5b0e.png)

**Expected results:** 
the container should have normal matrix4 perspective effect

**Actual results:** 
the effect is wierd

<details>
<summary>Code sample</summary>

```dart
Container(
                    width: imageWidth.first,
                    height: cardHeight[3],
                    decoration: ShapeDecoration(
                      shape: const RoundedRectangleBorder(
                        borderRadius: BorderRadius.all(Radius.circular(20)),
                      ),
                      color: const Color(0xFFFF9D9D).withOpacity(0.2),
                    ),
                    transform: Matrix4.identity()
                      ..setEntry(3, 2, 0.001)
                      ..rotateY(0.01)
                      //..translate(translateXList[3], translateYList[3]),
                  ),
```

</details>

<details>
  <summary>Logs</summary>

```sh
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.1, on macOS 13.0 22A380 darwin-x64, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] Android Studio (version 2021.3)
[!] Android Studio (version 2022.1)
    ✗ Unable to find bundled Java version.
[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
[✓] VS Code (version 1.75.0)
[✓] Connected device (4 available)
[✓] HTTP Host Availability
```

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue with `Master (3.8.0-10.0.pre.34)`.

<details><summary>Code Sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(title)),
      body: Center(
        child: Container(
          width: 300,
          height: 300,
          decoration: const ShapeDecoration(
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(20)),
            ),
            color: Colors.red,
          ),
          transform: Matrix4.identity()
            ..setEntry(3, 2, 0.001)
            ..rotateY(1),
        ),
      ),
    );
  }
}
```

</details>

<details><summary>Screenshots</summary>

#### Without impeller (Expected)

<img width="300" src="https://user-images.githubusercontent.com/50555895/218445882-958ac65f-137b-4106-b360-b7ffb17b1451.png">

#### With impeller (Actual)

<img width="300" src="https://user-images.githubusercontent.com/50555895/218445694-1f08214e-6413-4163-ae6b-5cd2c9873a94.png">

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.8.0-10.0.pre.34, on macOS 13.2 22D49 darwin-arm64, locale en-US)
    • Flutter version 3.8.0-10.0.pre.34 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 53fe8a3f94 (74 minutes ago), 2023-02-13 02:37:24 -0500
    • Engine revision 4107a7b71f
    • Dart version 3.0.0 (build 3.0.0-231.0.dev)
    • DevTools version 2.21.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.75.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 6BC61E88-4DE0-4BCB-8599-DA5294269146 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.2 22D49 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Brave Browser 110.1.48.158

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by chinmaygarde** [comment]
We think this is the same underlying issue as https://github.com/flutter/flutter/issues/120272. I'm closing this as a duplicate but have cross referenced it so that we can verify this case as well. Thanks for trying Impeller.
